// A very simple protocol definition, consisting of only
// one message.

syntax = "proto2";

package radarPackage;

message targetMessage {
    required uint32 id = 1;
    required int32 longitudinalX = 2;//longitudinalX = longitudinalX*1e7
    required int32 lateralY = 3;//lateralY = lateralY*1e7
	required int32 velocity = 4;		// 速度 km/h 小数点右移两位，0.01的精度 如：10.01km/h  传入1001
    required uint32 heading = 5;	//度*100
    required int32 RCS = 6;//rcs = rcs*100
}

message Timestamp {
  // Represents seconds of UTC time since Unix epoch
  // 1970-01-01T00:00:00Z. Must be from 0001-01-01T00:00:00Z to
  // 9999-12-31T23:59:59Z inclusive.
  required int64 seconds = 1;

  // Non-negative fractions of a second at nanosecond resolution. Negative
  // second values with fractions must still have non-negative nanos values
  // that count forward in time. Must be from 0 to 999,999,999
  // inclusive.
  required int32 nanos = 2;
}

message radarMsg {
    required Timestamp time = 1;
    repeated targetMessage msg = 2; 
}
